src/main/java/uk/gov/pay/api/service/GetDirectDebitPaymentService.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.pay.api.service;
+
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.exception.GetChargeException;
+import uk.gov.pay.api.model.directdebit.DirectDebitConnectorPaymentResponse;
+import uk.gov.pay.api.model.directdebit.mandates.DirectDebitPayment;
+
+import javax.inject.Inject;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+
+import static org.apache.http.HttpStatus.SC_OK;
+import static uk.gov.pay.api.model.directdebit.mandates.DirectDebitPayment.DirectDebitPaymentBuilder.aDirectDebitPayment;
+
+public class GetDirectDebitPaymentService {
+
+    private final Client client;
+    private final PublicApiUriGenerator publicApiUriGenerator;
+    private final ConnectorUriGenerator connectorUriGenerator;
+
+    @Inject
+    public GetDirectDebitPaymentService(Client client, PublicApiUriGenerator publicApiUriGenerator, ConnectorUriGenerator connectorUriGenerator) {
+        this.client = client;
+        this.publicApiUriGenerator = publicApiUriGenerator;
+        this.connectorUriGenerator = connectorUriGenerator;
+    }
+
+    public DirectDebitPayment getDirectDebitPayment(Account account, String paymentId) {
+        Response connectorResponse = client
+                .target(connectorUriGenerator.chargeURI(account, paymentId))
+                .request()
+                .get();
+        if (connectorResponse.getStatus() != SC_OK) {
+            throw new GetChargeException(connectorResponse);
+        }
+        DirectDebitConnectorPaymentResponse ddConnectorResponse = 
+                connectorResponse.readEntity(DirectDebitConnectorPaymentResponse.class);
+        return aDirectDebitPayment()
+                    .withDescription(ddConnectorResponse.getDescription())
+                    .withAmount(ddConnectorResponse.getAmount())
+                    .withPaymentId(ddConnectorResponse.getPaymentExternalId())
+                    .withPaymentProvider(ddConnectorResponse.getPaymentProvider())
+                    .withState(ddConnectorResponse.getState())
+                    .withCreatedDate(ddConnectorResponse.getCreatedDate())
+                    .withReference(ddConnectorResponse.getReference())
+                    .withSelfLink(publicApiUriGenerator.getDirectDebitPaymentURI(paymentId))
+                    .withMandateLink(publicApiUriGenerator.getMandateURI(ddConnectorResponse.getMandateId()))
+                    .withEventsLink(publicApiUriGenerator.getMandateEventsURI(ddConnectorResponse.getMandateId()))
+                    .build();
```

Comment:
Because we now do this in 2 places, I reckon it's probably worthwhile to have a `from(DirectDebitConnectorPaymentResponse paymentResponse)` method on `DirectDebitPayment` to avoid the duplication.
You can do this in another PR if you prefer